### PR TITLE
Allow additional attributes to be added to client spans

### PR DESCRIPTION
**Pull request by bpholt**
I hope to use this to add Otel attributes like `rpc.system` and `peer.service` to requests.

### CODE REVIEW

File: modules/http4s/src/main/scala/natchez/http4s/NatchezMiddleware.scala [hunk 1]
```diff
@@ -93,9 +93,42 @@ object NatchezMiddleware {
...
+   * - "client.http.uri"         -> request URI
+   * - "client.http.status_code" -> "200", "403", etc. // why is this a string?
+   *
+   * @param client the `Client[F]` to be enhanced
+   * @param additionalAttributes a function that takes the `Request[F]` and returns any additional attributes to be added to the span
+   * @tparam F An effect with instances of `Trace[F]` and `MonadCancelThrow[F]`
+   * @return the enhanced `Client[F]`
+   */
+  def client[F[_] : Trace : MonadCancelThrow](client: Client[F],
+                                              additionalAttributes: Request[F] => Seq[(String, TraceValue)],
```
Comment (armanbilge) [inline modules/http4s/src/main/scala/natchez/http4s/NatchezMiddleware.scala:130]: Thoughts on accepting an effectual method here?

```suggestion
                                              additionalAttributes: Request[F] => F[Seq[(String, TraceValue)]],
```

Comment (bpholt) [reply modules/http4s/src/main/scala/natchez/http4s/NatchezMiddleware.scala:130]: Yes, that's a great idea. I'll update the PR accordingly.

Comment (bpholt) [reply modules/http4s/src/main/scala/natchez/http4s/NatchezMiddleware.scala]: @armanbilge This has been updated if you want to take another look.

**Comment by armanbilge** [comment]
@bpholt sorry, this one got conflicty ...

**Comment by bpholt** [comment]
No worries, I'll resolve. I have another change that should be added too.

**Pull request merged**